### DISCUSSION TITLE
Compatibility with `black`

### DISCUSSION THREAD

**Discussion by evan0greenup** [astral-sh/ruff#10138] [Q&A] [answered]
I previously used `black` for code formatting. But, I want to switch to `ruff format` for speed.

However, I don't know if the formatting result from `ruff format` is compatible with `black`.

If not, is there any way to configure in `pyproject.toml` to make their result consistent?

**Comment by MichaReiser** [comment, marked as answer]
Hy. 

I hope the text from our website helps to understand the compatibility: 

> As such, the formatter is designed as a drop-in replacement for [Black](https://github.com/psf/black), but with an excessive focus on performance and direct integration with Ruff. Given Black's popularity within the Python ecosystem, targeting [Black compatibility](https://docs.astral.sh/ruff/formatter/#black-compatibility) ensures that formatter adoption is minimally disruptive for the vast majority of projects.

> Specifically, the formatter is intended to emit near-identical output when run over existing Black-formatted code. When run over extensive Black-formatted projects like Django and Zulip, > 99.9% of lines are formatted identically. (See: [Black compatibility
(https://docs.astral.sh/ruff/formatter/#black-compatibility).)

> Given this focus on Black compatibility, the formatter thus adheres to [Black's (stable) code style](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html), which aims for "consistency, generality, readability and reducing git diffs". To give you a sense for the enforced code style, here's an example:

[Source](https://docs.astral.sh/ruff/formatter/#philosophy)

One thing to note is that we haven't shipped the Black 24 style changes yet (we plan to do it this month). So there might be a few more changes when migrating from Black to ruff 0.2.x but there should only be very few changes after we released ruff 0.3
